### PR TITLE
Remove CaseClause related Operation kinds which are already represent…

### LINKED ISSUE #22129: OperationKind.XXXCaseClause and enum CaseKind have duplication
We decided in an earlier design meeting that we will have a single `OperationKind.CaseClause` for all case clauses, with a separate CaseKind enum to differentiate different kind of case clauses.

### PR DESCRIPTION
**Pull request by mavasani**
…ed through CaseKind

I also found that Range and Relational case clause have no tests in VB. Additionally, I also marked all OperationAnalyzerTests with CompilerTrait attribute so they don't get skipped in local trait based testing.

Fixes #22129

### CODE REVIEW

**Comment by 333fred** [comment]
Are you sure we didn't decide to remove CaseKind? I seem to recall us deciding that because they were all different nodes, though I could be mistaken.

**Comment by mavasani** [comment]
@333fred I can't completely recall and I couldn't find any related notes. @dotnet/analyzer-ioperation, we can probably decide the preferred API on this PR I can update accordingly. Fred's argument does sound reasonable to me.

File: src/Compilers/VisualBasic/Test/Semantic/Semantics/SelectCaseTests.vb [hunk 1]
```diff
@@ -410,6 +410,56 @@ End Module
...
+            Case Is < j'BIND:"Is < j"
+                Return True
+            Case Else
+                Return False
+        End Select
+    End Function
+End Class]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IRelationalCaseClause (Relational operator kind: BinaryOperatorKind.LessThan) (CaseKind.Relational) (OperationKind.RelationalCaseClause) (Syntax: 'Is < j')
```
Comment (cston) [inline src/Compilers/VisualBasic/Test/Semantic/Semantics/SelectCaseTests.vb]: Is `OperationKind.RelationalCaseClause` correct?

**Review by cston — APPROVED** [review]

File: src/Compilers/VisualBasic/Test/Semantic/Semantics/SelectCaseTests.vb (hunk 1 above)
Comment (mavasani) [reply src/Compilers/VisualBasic/Test/Semantic/Semantics/SelectCaseTests.vb]: No, the test needs to be fixed - I wanted to wait for everyone's review before fixing the test. Thanks.

**Comment by AlekseyTs** [comment]
@mavasani It looks like there are legitimate test failures.

**Comment by mavasani** [comment]
@AlekseyTs - yes one test is a legitimate failure. I was planning to fix it as soon as we have consensus on the API.

File: src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs [hunk 2]
```diff
@@ -4018,7 +4018,7 @@ public LazyReturnStatement(OperationKind kind, Lazy<IOperation> returnedValue, S
     internal abstract partial class BaseSingleValueCaseClause : CaseClause, ISingleValueCaseClause
     {
         public BaseSingleValueCaseClause(CaseKind caseKind, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
-            base(caseKind, OperationKind.SingleValueCaseClause, semanticModel, syntax, type, constantValue, isImplicit)
+            base(caseKind, semanticModel, syntax, type, constantValue, isImplicit)
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs]: >caseKind [](start = 17, length = 8)

What different case kinds can be used with this node?

File: src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs [hunk 3]
```diff
@@ -3797,7 +3797,7 @@ public override void Accept(OperationVisitor visitor)
     internal abstract partial class BaseRangeCaseClause : CaseClause, IRangeCaseClause
     {
         public BaseRangeCaseClause(CaseKind caseKind, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
-            base(caseKind, OperationKind.RangeCaseClause, semanticModel, syntax, type, constantValue, isImplicit)
+            base(caseKind, semanticModel, syntax, type, constantValue, isImplicit)
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs]: >caseKind [](start = 17, length = 8)

What different case kinds can be used with this node?

File: src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs [hunk 4]
```diff
@@ -3872,7 +3872,7 @@ internal sealed partial class LazyRangeCaseClause : BaseRangeCaseClause, IRangeC
     internal abstract partial class BaseRelationalCaseClause : CaseClause, IRelationalCaseClause
     {
         public BaseRelationalCaseClause(BinaryOperatorKind relation, CaseKind caseKind, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
-            base(caseKind, OperationKind.RelationalCaseClause, semanticModel, syntax, type, constantValue, isImplicit)
+            base(caseKind, semanticModel, syntax, type, constantValue, isImplicit)
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs]: >caseKind [](start = 17, length = 8)

What different case kinds can be used with this node?

Comment (AlekseyTs) [reply src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs]: It looks like all CaseKinds can be hardcoded.

---
In reply to: [139253449](https://github.com/dotnet/roslyn/pull/22133#discussion_r139253449) [](ancestors = 139253449)

**Review by AlekseyTs — APPROVED** [review]
LGTM

**Comment by mavasani** [comment]
VS Integration failures are known issues.

**Pull request merged**